src/javascript/detectors/session_fixation/session_fixation.js

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+// {fact rule=session-fixation@v1.0 defects=1}
+var express = require('express')
+var passport = require('passport')
+var app = express()
+function sessionFixationNoncompliant() {
+    app.post('/somepage',
+        passport.authenticate('local', { failureRedirect: '/somepage' }),
+        function(req, res) {
+            // Noncompliant: session.regenerate is absent.
+            res.redirect('/')
+        })
+}
+// {/fact}
+
+
+// {fact rule=session-fixation@v1.0 defects=0}
+var express = require('express')
+var passport = require('passport')
+var app = express()
+function sessionFixationCompliant() {
+    app.post('/somepage',
+        passport.authenticate('local', { failureRedirect: '/somepage' }),
+        function(req, res) {
+            // Compliant: session.regenerate is used
+            req.session.regenerate((err) => {
```

Comment:
Do we want to also call the redirect to keep it consistent with the example above?

Reply:
I think the redirection for the error scenario should be the error path or 404 path (`/error` or `/404`) .